tensorflow/compiler/jit/xla_compilation_cache.cc

```diff
@@ -71,6 +71,8 @@ XlaCompilationCache::~XlaCompilationCache() {
                     "programs to complete";
     }
   }
+  // Wait for all outstanding compilations to finish.
```

Comment:
Add a note here why we can't just rely on the `std::unique_ptr` auto-destructing.  Otherwise I'm pretty sure we'll get a helpful "cleanup fix NFC" few months down that road that will reintroduce the bug by deleting the "unnecessary" `.reset()`. :)

Reply:
done